spring-boot-project/spring-boot/src/main/java/org/springframework/boot/jdbc/metadata/DataSourcePoolMetadata.java

```diff
@@ -79,4 +80,13 @@
 	 */
 	Boolean getDefaultAutoCommit();
 
+	/**
+	 * Return the number of established but idle connections. Can also return {@code null}
+	 * if that information is not available.
+	 * @return the number of established but idle connections or {@code null}
+	 */
+	default Integer getIdle() {
```

Comment:
Rather than adding this method at the end I'd group it logically with a similar concept. Right below `getActive` sounds good. As a result, methods in implementations and tests should be reordered accordingly.